From 40882553e8210dae0736c0bcb9d5d39d50ccdcfe Mon Sep 17 00:00:00 2001
From: Klee Dienes <klee@mit.edu>
Date: Sat, 19 Mar 2016 07:21:52 -0400
Subject: [PATCH] Add 'extern' keyword to hidrd_natv_sink.

Otherwise the 'const' keyword gives it wonky linkage.  With this change,
hidrd_natv_sink matches hidrd_natv_src.
---
 include/hidrd/fmt/natv/snk.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/hidrd/fmt/natv/snk.h b/include/hidrd/fmt/natv/snk.h
index 5d043bd..4b367ef 100644
--- a/include/hidrd/fmt/natv/snk.h
+++ b/include/hidrd/fmt/natv/snk.h
@@ -34,7 +34,7 @@ extern "C" {
 #endif
 
 /** Native sink type */
-const hidrd_snk_type    hidrd_natv_snk;
+extern const hidrd_snk_type hidrd_natv_snk;
 
 /** Native sink error code */
 typedef enum hidrd_natv_snk_err {

From b622af32bd0e72cf0c6dffff6c9c8852f065051e Mon Sep 17 00:00:00 2001
From: Nikolai Kondrashov <Nikolai.Kondrashov@redhat.com>
Date: Sat, 18 Jun 2016 19:20:31 +0300
Subject: [PATCH] Add missing extern to declarations

Add "extern" to several missing declarations throughout the code, in
particular to avoid uninitialized copies of variables. This fixes an
assertion failure upon "hidrd-convert -h" invocation.
---
 include/hidrd/fmt/code/snk.h            | 2 +-
 include/hidrd/fmt/spec/snk.h            | 2 +-
 include/hidrd/usage/id_desc_list.h.m4   | 4 ++--
 include/hidrd/usage/page_desc_list.h.m4 | 4 ++--
 5 files changed, 7 insertions(+), 7 deletions(-)

diff --git a/include/hidrd/fmt/code/snk.h b/include/hidrd/fmt/code/snk.h
index e595f64..4d826fe 100644
--- a/include/hidrd/fmt/code/snk.h
+++ b/include/hidrd/fmt/code/snk.h
@@ -34,7 +34,7 @@ extern "C" {
 #endif
 
 /** Source code sink type */
-const hidrd_snk_type    hidrd_code_snk;
+extern const hidrd_snk_type hidrd_code_snk;
 
 /** Source code sink instance */
 typedef struct hidrd_code_snk_inst {
diff --git a/include/hidrd/fmt/spec/snk.h b/include/hidrd/fmt/spec/snk.h
index 56974f4..ca073d4 100644
--- a/include/hidrd/fmt/spec/snk.h
+++ b/include/hidrd/fmt/spec/snk.h
@@ -51,7 +51,7 @@ extern bool hidrd_spec_snk_flush(hidrd_snk *snk);
 extern void hidrd_spec_snk_clnp(hidrd_snk *snk);
 
 /** Specification example sink type */
-const hidrd_snk_type    hidrd_spec_snk;
+extern const hidrd_snk_type hidrd_spec_snk;
 
 /** Specification example sink item state table */
 typedef struct hidrd_spec_snk_state hidrd_spec_snk_state;
diff --git a/include/hidrd/usage/id_desc_list.h.m4 b/include/hidrd/usage/id_desc_list.h.m4
index bc57082..1cde027 100644
--- a/include/hidrd/usage/id_desc_list.h.m4
+++ b/include/hidrd/usage/id_desc_list.h.m4
@@ -68,13 +68,13 @@ extern "C" {
 extern bool hidrd_usage_id_desc_list_valid(const hidrd_usage_id_desc *list,
                                            size_t                     num);
 
-const hidrd_usage_id_desc *hidrd_usage_id_desc_list_lkp_by_value(
+extern const hidrd_usage_id_desc *hidrd_usage_id_desc_list_lkp_by_value(
                                         const hidrd_usage_id_desc  *list,
                                         size_t                      num,
                                         hidrd_usage_id              value);
 
 #ifdef HIDRD_WITH_TOKENS
-const hidrd_usage_id_desc *hidrd_usage_id_desc_list_lkp_by_token(
+extern const hidrd_usage_id_desc *hidrd_usage_id_desc_list_lkp_by_token(
                                         const hidrd_usage_id_desc  *list,
                                         size_t                      num,
                                         const char                 *token);
diff --git a/include/hidrd/usage/page_desc_list.h.m4 b/include/hidrd/usage/page_desc_list.h.m4
index 7331d56..f83ebd1 100644
--- a/include/hidrd/usage/page_desc_list.h.m4
+++ b/include/hidrd/usage/page_desc_list.h.m4
@@ -80,7 +80,7 @@ popdef(`page_num')dnl
  *
  * @return Page description or NULL, if not found.
  */
-const hidrd_usage_page_desc *hidrd_usage_page_desc_list_lkp_by_value(
+extern const hidrd_usage_page_desc *hidrd_usage_page_desc_list_lkp_by_value(
                                                 hidrd_usage_page    value);
 
 #ifdef HIDRD_WITH_TOKENS
@@ -91,7 +91,7 @@ const hidrd_usage_page_desc *hidrd_usage_page_desc_list_lkp_by_value(
  *
  * @return Page description or NULL, if not found.
  */
-const hidrd_usage_page_desc *hidrd_usage_page_desc_list_lkp_by_token(
+extern const hidrd_usage_page_desc *hidrd_usage_page_desc_list_lkp_by_token(
                                                 const char         *token);
 #endif /* HIDRD_WITH_TOKENS */
 
